content/blog/2022-01-15-moving-local-experiments-to-the-cloud-with-iterative-terraform-provider.md

```diff
@@ -0,0 +1,305 @@
+can take advantage of an
+[existing pre-configured Deep Learning AMI](https://docs.aws.amazon.com/dlami/latest/devguide/options.html).
+In this tutorial, we'll use an AMI, as it is the quickest way to start using
+GPUs on an AWS EC2 machine. An AMI can be selected using the `image` argument
+using the format `{user}@{owner}:{architecture}:{name}`. See the official
+[AMI documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html)
+to learn more.
+
+```hcl
+...
+resource "iterative_task" "tpi-examples-gpu" {
+  name      = "ami-gpu-example"
+  cloud     = "aws"
+  region    = "us-east-2"
+  machine   = "l+k80"
+  disk_size = "130"
+  # see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html for images
+  image     = "ubuntu@898082745236:x86_64:Deep Learning AMI (Ubuntu 18.04) Version 54.0"
+
+  workdir { input = "." }
+  script = <<-END
+    #!/bin/bash
+    pip3 install -r requirements.txt
+    python3 src/train.py
+  END
+}
+```
+
+The Deep Learning image that you choose may require larger disk size than the
+provided default, so you may need to increase it. The image I picked expects a
+volume of at least 130 GB, and I can configure this in the `disk_size` argument.
+
+You may also notice that my script got a bit shorter. This is because I no
+longer need to install Python, and only need to make sure to have my
+requirements met.
+
+## Ready… Set… Apply!
+
+Whether you want to go with the basic example, or the GPU-enabled training, you
+can run
+
+```
+terraform plan
+```
+
+to review what steps Terraform is going to take to make your desired
+infrastructure materialize. Don't worry, nothing is actually done at this point,
+but it's a good way to check for potential issues in the configuration.
+
+If there are no issues, run
+
+```
+terraform apply
+```
+
+to execute the actions proposed in the Terraform plan. You'll need to type `yes`
+to confirm that you indeed would like the plan to be executed.
+
+At this point you can get a cup of your favourite beverage, and let the
+Iterative Provider work its magic together with Terraform. They will allocate a
+remote machine for you, set up the environment and run your `train.py`. Once the
+script finishes, the machine will be terminated.
+
+You can monitor what's going on by running
+
+```
+terraform refresh && terraform show
+```
+
+Once you see that the task has successfully finished, go ahead and run
+
+```
+terraform destroy
+```
+
+to destroy all remote objects managed by your configuration and sync back your
+shared files. If you write metrics to a file (e.g. `metrics.json`), then you'll
+get the new metrics synced back to your local machine.
+
+Now if you want to try another experiment, you can change your code, run
+`terraform apply` again, and when the training is finished, commit your code
+together with the updated `metrics.json`. This can help you move from
+prototyping locally to leveraging more powerful cloud machines without the
```

Comment:
🤖 🤦🏼‍♂️

Reply:
> _[The reason for the unreason to which my reason is subjected, so weakens my reason that I have reason to complain of your beauty.](https://es.wikisource.org/wiki/Don_Quijote,_Primera_Parte:_Cap%C3%ADtulo_I)_

Yes, too many machines to explain a simple task. There are machines involved, of course, but we can merge the duplicate ones to avoid confusing users with a thousand different machine [epithets](https://en.m.wikipedia.org/wiki/Category:Epithets_of_Greek_deities).